File: 15May-2026/Day1/find_duplicates.py
import random


def generate_random_numbers(size=20, start=1, end=15):
    """Generate a list of random numbers (with possible duplicates)."""
    return [random.randint(start, end) for _ in range(size)]


def find_duplicates(numbers):
    """Find and return duplicate numbers from a list."""
    seen = set()
    duplicates = set()

    for num in numbers:
        if num in seen:
            duplicates.add(num)
        else:
            seen.add(num)

    return sorted(duplicates)


if __name__ == "__main__":
    numbers = generate_random_numbers()
    print(f"Generated numbers : {numbers}")

    duplicates = find_duplicates(numbers)

    if duplicates:
        print(f"Duplicate numbers : {duplicates}")
    else:
        print("No duplicates found.")

File: 15May-2026/Day1/findDuplicates.js
/**
 * Generate an array of random numbers (with possible duplicates).
 * @param {number} size  - how many numbers to generate
 * @param {number} start - minimum value (inclusive)
 * @param {number} end   - maximum value (inclusive)
 * @returns {number[]}
 */
function generateRandomNumbers(size = 20, start = 1, end = 15) {
  return Array.from({ length: size }, () =>
    Math.floor(Math.random() * (end - start + 1)) + start
  );
}

/**
 * Find duplicate numbers in an array.
 * @param {number[]} numbers
 * @returns {number[]} sorted list of duplicates
 */
function findDuplicates(numbers) {
  const seen = new Set();
  const duplicates = new Set();

  for (const num of numbers) {
    if (seen.has(num)) {
      duplicates.add(num);
    } else {
      seen.add(num);
    }
  }

  return [...duplicates].sort((a, b) => a - b);
}

// --- main ---
const numbers = generateRandomNumbers();
console.log("Generated numbers :", numbers);

const duplicates = findDuplicates(numbers);

if (duplicates.length > 0) {
  console.log("Duplicate numbers :", duplicates);
} else {
  console.log("No duplicates found.");
}
